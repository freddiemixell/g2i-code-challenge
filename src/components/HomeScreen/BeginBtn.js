import React, { useContext } from 'react';
import { Link } from 'react-router-dom';

import { QuizContext } from '../QuizProvider/QuizProvider';

export default function BeginBtn() {
    let { resetQuiz } = useContext( QuizContext );

    return (
        <Link
            className="button"
            to="/quiz"
            onClick={ () => resetQuiz() }
        >
            Begin
        </Link>
    );
}