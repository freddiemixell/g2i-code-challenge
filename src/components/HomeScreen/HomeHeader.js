import React from 'react';

export default function HomeHeader() {
    return (
        <>
            <div className="container">
                <h1 className="title">Welcome to the Trivia Challenge!</h1>
            </div>
            <div className="container">
                <p>You will be presented with 10 true or false questions.</p>
            </div>
            <div className="container">
                <p>Can you score 100%?</p>
            </div>
        </>
    );
}