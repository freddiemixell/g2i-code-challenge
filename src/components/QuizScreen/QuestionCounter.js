import React, { useContext } from 'react';

import { Container } from '../../utils/elements';
import { QuizContext } from '../QuizProvider/QuizProvider';

export default function QuestionCounter() {
    let { currentQuestion = 0 } = useContext( QuizContext );
    return (
        <Container>
            <p>{ currentQuestion + 1 } of 10</p>
        </Container>
    );
}
