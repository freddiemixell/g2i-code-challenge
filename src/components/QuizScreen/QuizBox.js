import React from 'react';

export default function QuizBox() {
    return (
        <>
            <div className="container">
                <div className="question">
                    <p>Unturned originally started as a Roblox game.</p>
                </div>
            </div>
            <div className="container">
                <div className="answer">
                    <button className="answer__btn answer__btn-true">True</button>
                    <button className="answer__btn answer__btn-false">False</button>
                </div>
            </div>
        </>
    );
}