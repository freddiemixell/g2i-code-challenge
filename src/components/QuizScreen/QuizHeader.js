import React, { useContext } from 'react';

import { QuizContext } from '../QuizProvider/QuizProvider';
import { Container, Title } from '../../utils/elements';

export default function QuizHeader() {
    let { questions = [{category: ''}], currentQuestion = 0 } = useContext( QuizContext );

    let title = typeof questions[currentQuestion] !== 'undefined' ? questions[currentQuestion].category : '';
    return title ? (
        <Container>
            <Title>{ title }</Title>
        </Container>
    ) : null;
}
