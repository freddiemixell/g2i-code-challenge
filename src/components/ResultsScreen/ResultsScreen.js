import React from 'react';

import ResultsHeader from './ResultsHeader';
import ScoreCard from './ScoreCard';
import PlayAgainBtn from './PlayAgainBtn';

export default function ResultsScreen() {
    return (
        <div className="section">
            <ResultsHeader/>
            <ScoreCard/>
            <PlayAgainBtn/>
        </div>
    );
}