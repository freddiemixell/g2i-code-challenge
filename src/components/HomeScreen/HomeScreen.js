import React from 'react';
import BeginBtn from './BeginBtn';
import HomeHeader from './HomeHeader';

import { Wrapper } from '../../utils/elements';

export default function HomeScreen() {
    return (
        <Wrapper>
            <HomeHeader/>
            <BeginBtn/>
        </Wrapper>
    );
}