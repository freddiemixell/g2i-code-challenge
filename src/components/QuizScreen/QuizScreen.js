import React, { useContext } from 'react';
import { FadeLoader } from 'react-spinners';

import QuizHeader from './QuizHeader';
import QuizBox from './QuizBox';
import QuestionCounter from './QuestionCounter';
import { Wrapper } from '../../utils/elements';
import { QuizContext } from '../QuizProvider/QuizProvider';

export default function QuizScreen() {
    let { isLoading } = useContext( QuizContext );

    return isLoading ? (
        <Loading/>
    ) : (
        <Wrapper>
            <QuizHeader/>
            <QuizBox/>
            <QuestionCounter/>
        </Wrapper>
    );
}

function Loading() {
    return (
        <Wrapper>
            <div style={{display:'flex', width: '100vw', height: '100vh', justifyContent: 'center', alignItems: 'center'}}>
                <FadeLoader
                    height={15}
                    width={5}
                    radius={2}
                    margin={2}
                />
            </div>
        </Wrapper>
    );
}