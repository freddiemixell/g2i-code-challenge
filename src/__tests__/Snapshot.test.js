import React from "react";
import Enzyme, { shallow, } from "enzyme";
import toJson from 'enzyme-to-json';
import Adapter from "enzyme-adapter-react-16";

import HomeScreen from '../components/HomeScreen/HomeScreen';
import QuizScreen from '../components/QuizScreen/QuizScreen';
import ResultsScreen from '../components/ResultsScreen/ResultsScreen';


Enzyme.configure( { adapter: new Adapter() } );

describe( 'Snapshots', () => {
    it( "renders correctly", () => {
        const home = shallow( <HomeScreen /> );
        const quiz = shallow( <QuizScreen/> );
        const results = shallow( <ResultsScreen/> );

        expect( toJson( home ) ).toMatchSnapshot();
        expect( toJson( quiz ) ).toMatchSnapshot();
        expect( toJson( results ) ).toMatchSnapshot();
    } );
} );
