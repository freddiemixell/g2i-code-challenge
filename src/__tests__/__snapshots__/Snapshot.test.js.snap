// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders correctly 1`] = `
<styled.section>
  <HomeHeader />
  <BeginBtn />
</styled.section>
`;

exports[`Snapshots renders correctly 2`] = `
<styled.section>
  <QuizHeader />
  <QuizBox />
  <QuestionCounter />
</styled.section>
`;

exports[`Snapshots renders correctly 3`] = `
<styled.section>
  <ResultsHeader />
  <ScoreCard />
  <PlayAgainBtn />
</styled.section>
`;
